src/components/VideoCard.jsx

```diff
@@ -0,0 +1,58 @@
+import { useNavigate } from "react-router-dom";
+import { v4 as uuid } from "uuid";
+import { useVideos } from "hooks";
+
+export function VideoCard() {
+  const navigate = useNavigate();
+  const { videoList } = useVideos();
+
+  return (
+    <div className="flex flex-row flex-wrap p-8">
+      {videoList.map(
+        ({ id, title, creator, views, uploadedOn, about, img, avatar }) => (
+          <div
+            className="h-96 w-1/4 p-3"
+            key={uuid()}
+            onClick={() => {
+              navigate(`/video/${id}`, {
+                state: {
+                  id,
```

Comment:
put into your global state as a variable "currentlyViewingVideo", also handle a case when page refreshes, you make an api call and store data in currentlyViewingVideo